src/pyats/contrib/plugins/webex.py

```diff
@@ -0,0 +1,170 @@
+                         default = None,
+                         help='Email of specific user to send WebEx '
+                              'notification to')
+        return grp
+
+
+    def post_job(self, job):
+        # Get WebEx info from arguments or configuration
+        token = self.runtime.args.webex_token or cfg.get('webex.token')
+        space = self.runtime.args.webex_space or cfg.get('webex.space')
+        email = self.runtime.args.webex_email or cfg.get('webex.email')
+
+        if not token:
+            logger.info('WebEx Token not given as argument or in config. No '
+                        'WebEx notification will be sent')
+            return
+
+        headers = {'Authorization': 'Bearer {}'.format(token),
+                   'Content-Type': 'application/json'}
+
+        if not space and not email:
+            logger.info('No Space ID or email specified, No WebEx Teams '
+                        'notification will be sent')
+            return
+
+        # Format message with info from job run
+        msg = MESSAGE_TEMPLATE.format(job=job)
+
+        try:
+            # Attempt to get path for TRADe logs
+            if not self.runtime.runinfo.no_upload:
+                msg += '\n\nView TRADe logs at: %s'\
+                       % self.runtime.runinfo.log_url
+        except AttributeError:
+            pass
+
+        try:
+            host = job.runtime.env['host']['name']
+            # Determine if liveview is running
+            if self.runtime.args.liveview and\
+                    self.runtime.args.liveview_keepalive:
+                # Liveview will set this to the assigned port if not specified
+                port = self.runtime.args.liveview_port
+                try:
+                    # Attempt to add a link using the host domain name
+                    addr = socket.getfqdn()
+                    socket.gethostbyname(addr)
+                    msg += '\n\nLogs can be viewed with the pyATS Log Viewer '\
+                           'at: http://%s:%s' % (addr, port)
+                except OSError:
+                    msg += '\n\nLogs can be viewed in your browser by '\
+                           'connecting to %s with port %s' % (host, port)
+            else:
+                # Show command to run liveview
+                archive = self.runtime.runinfo.archive_file
+                if archive:
+                    msg += '\n\nRun the following command on %s to view logs '\
+                           'from this job: `pyats logs view %s --host 0.0.0.0`'\
+                           % (host, archive)
+
+        except AttributeError:
+            pass
+
+        # Build payload
+        payload = {'markdown': msg}
+        if space:
+            payload['roomId'] = space
+        elif email:
+            payload['toPersonEmail'] = email
+
+        logger.info('Sending WebEx Teams notification')
+
+        try:
+            # Attempt POST
+            r = requests.post(MESSAGE_URL,
```

Comment:
This is REST. why not develop rest.connector for webex teams and call it here? with the rest.connector, we can have device in testbed yaml and also we can send notification even during runs and in testcases and anytime when user wants.